cpp/daal/src/algorithms/covariance/covariance_impl.i

```diff
@@ -135,8 +136,14 @@ static inline size_t getBlockSize(size_t nrows)
     return 140;
 }
 
+#ifdef __ARM_ARCH
+#define CPU_TYPE sve
+#else
+#define CPU_TYPE avx512
+#endif
+
 template <>
-inline size_t getBlockSize<avx512>(size_t nrows)
+inline size_t getBlockSize<CPU_TYPE>(size_t nrows)
 {
     return (nrows > 5000 && nrows <= 50000) ? 1024 : 140;
```

Comment:
Is this a good block size for SVE as well? There is already the default block size in the implementation above this. Maybe `#ifdef` this and include it only if AVX512 can be present, e.g. `#ifdef DAAL_KERNEL_AVX512`